### PR TITLE
REGISTRY-3139 Adding test cases for duration time in lifecycle at publisher

### PR DESCRIPTION
**Pull request by lahirue**

### CODE REVIEW

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+    @Factory(dataProvider = "userModeProvider")
+    public GRegPublisherLCStateDurationColorBoxTestCase(
+            TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    private static TestUserMode[][] userModeProvider() {
+        return new TestUserMode[][] { new TestUserMode[] { TestUserMode.SUPER_TENANT_ADMIN },
```
Comment (sameerak) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java]: indent properly
apply style sheet

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+                //                new TestUserMode[]{TestUserMode.TENANT_USER},
```
Comment (sameerak) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java]: remove commented out code

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    @Factory(dataProvider = "userModeProvider")
+    public GRegPublisherLCStateDurationTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    private static TestUserMode[][] userModeProvider() {
+        //This method use run in tenant mode
+        return new TestUserMode[][] { new TestUserMode[] { TestUserMode.SUPER_TENANT_ADMIN },
```
Comment (sameerak) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java]: indent code properly
apply style sheet

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                //                new TestUserMode[]{TestUserMode.TENANT_USER},
```
Comment (sameerak) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java]: remove commented out code

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java (hunk 1 above)
Comment (lahirue) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java]: Thank you Sameera. This code block has been refactored with new time in lifecycle tests

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java (hunk 2 above)
Comment (lahirue) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java]: Thank you Sameera for mentioning it. This code block has been refactored with new time in lifecycle tests

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java (hunk 3 above)
Comment (lahirue) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java]: Thank you Sameera. This code block has been refactored with new time in lifecycle tests

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java (hunk 4 above)
Comment (lahirue) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java]: Thank you Sameera. This code block has been refactored with new time in lifecycle tests

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+/***
+ * This class will test Greg Publisher Lifecycle State Duration and Its Color bellow the asset icon
+ */
+public class GRegPublisherLCStateDurationColorBoxTestCase extends GregESTestBaseTest {
+    public static final String CLASS_NAME = GRegPublisherLCStateDurationTestCase.class.getName();
+    private static final Log log = LogFactory.getLog(GRegPublisherLCStateDurationTestCase.class);
+    String assetId;
```
Comment (nuwanw) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java]: make class variable private

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+/***
+ * This Class will test Greg Publisher Lifecycle State Duration and Its Color inside Lifecycle Tab
+ */
+public class GRegPublisherLCStateDurationTestCase extends GregESTestBaseTest {
+    public static final String CLASS_NAME = GRegPublisherLCStateDurationTestCase.class.getName();
+    private static final Log log = LogFactory.getLog(GRegPublisherLCStateDurationTestCase.class);
+    String assetId;
```
Comment (nuwanw) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java]: make variable private

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java (hunk 5 above)
Comment (lahirue) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java]: Thank you Nuwan for mentioning this,
This entire code block has been refactored with new time in lifecycle feature.

File: modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java (hunk 6 above)
Comment (lahirue) [inline modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationTestCase.java]: Thank you Nuwan for mentioning ,
This entire code block has been refactored with new time in lifecycle feature.

**Comment by daneshk** [comment]
Closing this PR, since this PR is outdated according to the review comment. Please reopen if we need to add after fixing the review comments

**Pull request closed without merging**